File: cdm/cpp/system/physiology/SEHepaticSystem.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "system/physiology/SEHepaticSystem.h"

SEHepaticSystem::SEHepaticSystem(Logger* logger) : SESystem(logger)
{
  
}

SEHepaticSystem::~SEHepaticSystem()
{
  Clear();
}

void SEHepaticSystem::Clear()
{
  SESystem::Clear();

}

const SEScalar* SEHepaticSystem::GetScalar(const std::string& name)
{
  return nullptr;
}

void SEHepaticSystem::Load(const cdm::HepaticSystemData& src, SEHepaticSystem& dst)
{
  SEHepaticSystem::Serialize(src, dst);
}
void SEHepaticSystem::Serialize(const cdm::HepaticSystemData& src, SEHepaticSystem& dst)
{
  dst.Clear();
  
}

cdm::HepaticSystemData* SEHepaticSystem::Unload(const SEHepaticSystem& src)
{
  cdm::HepaticSystemData* dst = new cdm::HepaticSystemData();
  SEHepaticSystem::Serialize(src, *dst);
  return dst;
}
void SEHepaticSystem::Serialize(const SEHepaticSystem& src, cdm::HepaticSystemData& dst)
{
  
}

File: cdm/java/mil/tatrc/physiology/datamodel/system/physiology/SEHepaticSystem.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/
package mil.tatrc.physiology.datamodel.system.physiology;

import com.kitware.physiology.cdm.Physiology.HepaticSystemData;

import mil.tatrc.physiology.datamodel.properties.*;
import mil.tatrc.physiology.datamodel.system.SESystem;

public class SEHepaticSystem extends SEPhysiologySystem implements SESystem
{
  public SEHepaticSystem()
  {
    
  }

  public void reset()
  {    
    
  }

  public static void load(HepaticSystemData src, SEHepaticSystem dst)
  {        
    
  }
  public static HepaticSystemData unload(SEHepaticSystem src)
  {
    HepaticSystemData.Builder dst = HepaticSystemData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  protected static void unload(SEHepaticSystem src, HepaticSystemData.Builder dst)
  {    
    
  }

  
}
